### PR TITLE
I15 just sql patch files

### PR DESCRIPTION
**Pull request by chilts**

### CODE REVIEW

**Comment by chilts** [comment]
@rfk This should tighten things up. r?

File: index.js [hunk 1]
```diff
@@ -195,20 +197,21 @@ Patcher.prototype.readPatchFiles = function readPatchFiles(callback) {
...
-    }).bind(this))
-
     async.eachLimit(
       files,
       10,
       (function(filename, done) {
+
+        // We're still doing a check here, since the glob above could get filenames like 'patch-this-001-002.sql' where
+        // the prefix was set as 'patch or 'patch-foo-bar.sql' where the globs are not numeric.
         var m = filename.match(/-(\d+)-(\d+)\.sql$/)
```
Comment (rfk) [inline index.js]: Will this regex actually detect the `path-this-001-002.sql` case?

File: test/patches/ignored.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+This file is ignored, since we tell mysql-patcher to look for files with a particular filename.
```
Comment (rfk) [inline test/patches/ignored.txt]: Nice, and it may be worth adding some more complex edge-cases here as well, e.g. `patch-ignored-foo-bar.sql`.

File: index.js (hunk 1 above)
Comment (chilts) [inline index.js]: Yeah, I should probably check that out. :) Will add a couple more tests and which should satisfy your other comment below. Ta.

**Comment by chilts** [comment]
Some extra tests added. Changed the whole regex thing into something slightly nicer but slightly longer. `nicer + longer > shorter + complicated`.

File: test/patches/patch-003-004.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+This file is also ignore since it has the wrong extension (even though the basename look okay).
```
Comment (rfk) [inline test/patches/patch-003-004.txt]: s/ignore/ignored

**Comment by rfk** [comment]
Looks good, r+ with typo fix

**Comment by chilts** [comment]
Released to npm in v0.7.0.
